core/trino-main/src/main/java/io/trino/sql/planner/ExpressionInterpreter.java

```diff
@@ -1506,6 +1507,37 @@ protected Object visitSubscriptExpression(SubscriptExpression node, Object conte
             return invokeOperator(OperatorType.SUBSCRIPT, types(node.getBase(), node.getIndex()), ImmutableList.of(base, index));
         }
 
+        @Override
+        protected Object visitExtract(Extract node, Object context)
+        {
+            Object value = processWithExceptionHandling(node.getExpression(), context);
+            if (value == null || value instanceof Expression) {
+                return null;
```

Comment:
Why return null if the value is an Expression?

Reply:
Good catch. That's wrong.